sdk/core/Azure.Core.Experimental/tests/public/Azure.Core.Experimental.Tests.Public.csproj

```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
Is this a good way to add tests for Core types where we don't want InternalsVisibleTo?  I'm super open to approaches here.  @JoshLove-msft, @christothes, and thoughts on this?

Reply:
Do you have an example of something that requires it to test properly?

Reply:
There are a couple of categories.
			
One is that we have a lot of internal methods on JsonData, so in test projects where we have InternalsVisibleTo, it's easy to accidentally write a test to mimic a user scenario that calls an internal method when a library user wouldn't actually be able to call this.  Adding a test project that only has access to Azure.Core public APIs gives us compiler checks to prevent this.

The second category is where we're using dynamic APIs going through the DynamicMetaObject interface and a member is available internally but not publicly.  The DMO interface obfuscates access modifier rules and it's another case where it's easy to write something you think is available publicly that's not, and it's nice to have automated validation of this both now, and for future modifications to the type.

I'm working on getting a clear test that both compiles and fails in public tests but succeeds in internal tests.  I had one with JsonData was internal with public members, now that it is public with internal members I'm having a harder time finding one, but trying to pin down enough specifics to do this.  I hope I can get this before next week when I'm back from jury duty so we can move forward on the approach in the interim.

@christothes can you share your specific concerns with this approach and what information might motivate you to invest in further thinking in this area?

Reply:
Thanks @JoshLove-msft for the help troubleshooting this!